module/zfs/dmu_objset.c

```diff
@@ -287,7 +287,8 @@ redundant_metadata_changed_cb(void *arg, uint64_t newval)
 	 * Inheritance and range checking should have been done by now.
 	 */
 	ASSERT(newval == ZFS_REDUNDANT_METADATA_ALL ||
-	    newval == ZFS_REDUNDANT_METADATA_MOST);
+	    newval == ZFS_REDUNDANT_METADATA_MOST ||
+	    newval == ZFS_REDUNDANT_METADATA_NONE);
```

Comment:
I think that we should implement this in a way that doesn't cause panics on older software (even on debug).  I see a few options for doing this:
1. Use a new key in the properties ZAP object (but ideally, keep the user interface the same).  So when getting the property we would have code something like:
```
potential_key = sprintf("%s$iuv", prop_name); // iuv = Ignore Unknown Values
if (zap_lookup(props_obj, potential_key, &potential_value) == SUCCESS &&
  property_value_is_understood(prop_name, potential_value)) {
    value = potential_value;
} else if (zap_lookup(props_obj, prop_name, &potential_value) == SUCCESS) {
    value = potential_value
} else {
    value = prop_get_default(prop_name);
}
```
This would provide infrastructure to add "unknown but ignorable" values to other properties in the future.
2. Add a new on-disk feature flag, so that once `redundant_metadata=none` is set, the pool can't be brought back to older systems.

Reply:
option 1 looks to be preferable, I can try something similar to the above code and get back with a fix.

Reply:
If the property is unknown how do you know the default value?

Reply:
Ah, if the property is known but the value is unknown.